src/main/ruby/truffleruby/core/string.rb

```diff
@@ -1490,12 +1490,22 @@ def rindex(sub, finish=undefined)
   end
 
   def start_with?(*prefixes)
+    binding = TrufflePrimitive.caller_binding
```

Comment:
Getting the caller binding can be expensive, could you get it only for the case where it's needed?
In this case, an easy way seems to be:
```ruby
  def start_with?(*prefixes)
    binding = TrufflePrimitive.caller_binding if prefixes.any?(Regexp)
```

Reply:
I'd simplify and just get the binding inside the block, but unfortunately `TrufflePrimitive.caller_binding` does not work inside blocks yet.

Reply:
@eregon, i've added the changes here: b4316a0